IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -876,4 +884,42 @@ - (BOOL)saveAccount:(MSIDAccount *)account
     return tokens;
 }
 
+- (BOOL)saveAppMetadataWithConfiguration:(MSIDConfiguration *)configuration
+                            response:(MSIDTokenResponse *)response
+                             context:(id<MSIDRequestContext>)context
+                               error:(NSError **)error
+{
+    MSIDAppMetadataCacheItem *metadata = [_factory appMetadataFromResponse:response configuration:configuration];
+    metadata.environment = [[configuration.authority cacheUrlWithContext:context] msidHostWithPortIfNecessary];
+    
+    if (metadata)
+    {
+        return [self saveAppMetadata:metadata context:context error:error];
+    }
+   
+    return YES;
+}
+
+- (BOOL)saveAppMetadata:(MSIDAppMetadataCacheItem *)metadata
```

Comment:
nit: those two methods seem quite short (saveAppMetadataWithConfiguration and saveAppMetadata). Can we combine them?

Reply:
done